Client/core/Graphics/CRenderItem.EffectCloner.h

```diff
@@ -19,12 +19,14 @@ class CEffectCloner
 public:
     CEffectCloner(CRenderItemManager* pManager);
     void         DoPulse();
-    CEffectWrap* CreateD3DEffect(const SString& strFile, const SString& strRootPath, bool bIsRawData, SString& strOutStatus, bool bDebug);
+    CEffectWrap* CreateD3DEffect(const SString& strFile, const SString& strRootPath, bool bIsRawData, SString& strOutStatus,
+        bool bDebug, const std::vector<std::pair<SString, SString>>& macros);
     void         ReleaseD3DEffect(CEffectWrap* pD3DEffect);
     void         MaybeTidyUp(bool bForceDrasticMeasures = false);
+    std::size_t  CalculateInvariantHash(const SString& strFile, const std::vector<std::pair<SString, SString>>& macros);
 
-    CElapsedTime                        m_TidyupTimer;
-    CRenderItemManager*                 m_pManager;
-    std::map<SString, CEffectTemplate*> m_ValidMap;            // Active and files not changed since first created
-    std::vector<CEffectTemplate*>       m_OldList;             // Active but files changed since first created
+    CElapsedTime                            m_TidyupTimer;
+    CRenderItemManager*                     m_pManager;
+    std::map<std::size_t, CEffectTemplate*> m_ValidMap;            // Active and files not changed since first created
```

Comment:
Please add a comment about `size_t` being a hash here, since its not clear at all.
I'd suggest to use `std::unordered_map` as it's faster (it also hash a hash function, which isn't much of a use here, sadly).